tools/osm/extract_env.py

```diff
@@ -10,66 +10,81 @@
+    parser.add_argument('-i', '--idir', type=str, help='Package installation directory')
+    parser.add_argument('-j', '--idireccodes', type=str, help='Eccodes installation directory. Default : same locatino as idir')
+    parser.add_argument('-f', '--force', action='store_true', help='Overwritte package installation if exists')
 
     args=parser.parse_args()
 
     if not args.spec:
-        print('Need a cosmo spack spec in order to extract the environement!')
+        sys.exit('Error: missing -s package spack spec in order to extract the environement!')
 
-    if not args.idircosmo:
-        print('Need an installation directory for installing cosmo and eccodes.')
+    if not args.idir:
+        sys.exit('Error: missing -i installation directory for installing the package and eccodes.')
     else:
-        args.idircosmo = os.path.abspath(args.idircosmo)
+        args.idir = os.path.abspath(args.idir)
+    
+    if not args.idireccodes:
+        args.idireccodes = args.idir  
 
     # Extract and concretize cosmo and eccodes
-    cosmo_spec=Spec(args.spec).concretized()
-
-    cosmo_dir = 'cosmo' + cosmo_spec.format('{@version}') + cosmo_spec.format('{%compiler}')  + '-' + cosmo_spec.format('{variants.real_type.value}') + '-' + cosmo_spec.format('{variants.cosmo_target.value}')
-    eccodes_dir = 'eccodes' + cosmo_spec.format('{^eccodes.@version}')
-    eccodes_definitions_dir = 'cosmo-eccodes-definitions' + cosmo_spec.format('{^cosmo-eccodes-definitions.@version}')
-
-    # Exist if cosmo installation already exists
-    if os.path.exists(os.path.join(args.idircosmo, cosmo_dir)):
-        sys.exit('Exit: The cosmo installation path: ' + os.path.join(args.idircosmo, cosmo_dir) + ' already exists!')
-    print('Installing ' + cosmo_spec.format('{prefix}') + ' to: ' + os.path.join(args.idircosmo, cosmo_dir))
-    print('COSMO_INSTALLATION_FULL_PATH=' + os.path.join(args.idircosmo, cosmo_dir))
-    os.system('cp -rf ' + cosmo_spec.format('{prefix}') + ' ' + os.path.join(args.idircosmo, cosmo_dir))
+    package_spec=Spec(args.spec).concretized()
+    print(package_spec.format('{prefix}'))
+
+    package_name = package_spec.format('{name}')
+    package_dir = package_name + package_spec.format('{@version}') + package_spec.format('{%compiler}')
+    if package_name == 'cosmo':
+        package_dir = package_dir + '-' + package_spec.format('{variants.real_type.value}') + '-' + package_spec.format('{variants.cosmo_target.value}')
+    eccodes_dir = 'eccodes' + package_spec.format('{^eccodes.@version}')
+    eccodes_definitions_dir = 'cosmo-eccodes-definitions' + package_spec.format('{^cosmo-eccodes-definitions.@version}')
+
+    if os.path.exists(os.path.join(args.idir, package_dir)):
+        if args.force:
+           # Remove if force cosmo installation already exists
+            print('Force : removing existing installation ' + os.path.join(args.idir, package_dir))
+            os.system('rm -r ' + os.path.join(args.idir, package_dir))
```

Comment:
```suggestion
          shutil.rmtree(os.path.join(args.idir, package_dir, ignore_errors=True))
```
might be overkil though